### DISCUSSION TITLE
QuadTree stack

### DISCUSSION THREAD

**Discussion by sinkinthesea** [jrouwe/JoltPhysics#1237] [Q&A] [answered]
I'm using jolt physics in a c# environment(Unity), and when i try to call RayCast Function, program crashed, and error is : Stack full.
I find that there is a const int "cStackSize = 128", I try to expand it, RayCast don't crash anymore, but ShapeCast still crash.

I don't know if i use Api by a wrong way, or is there someting should setting but i didn't set? Or the body in my environment should be less. 

![image](https://github.com/user-attachments/assets/ee25c075-381c-4ec0-a0de-c601ff653215)

**Comment by jrouwe** [comment, marked as answer]
You are triggering an 'assert' which means you're running a debug build and this normally triggers a breakpoint in the debugger. You should be able to continue after hitting the breakpoint, the code is guarded and won't crash (if it does then something is wrong in the C# bindings). Release builds don't trigger the assert. The only thing that will happen is that your raycast potentially misses collisions.

What probably happened is that you are adding a lot of bodies through `BodyInterface::AddBody` creating a very unbalanced broad phase tree. Try to batch add bodies using `BodyInterface::AddBodiesPrepare` followed by `BodyInterface::AddBodiesFinalize` or alternatively calling `PhysicsSystem::OptimizeBroadPhase` after adding many bodies through the `BodyInterface::AddBody` interface.

**Reply by sinkinthesea** [reply]
Thanks a lot, just as what you say, i build a release dll and it doesn't throw error any more; There are too many box in my physics world, and i'm trying to use OptimizeBroadPhase API. Thank you.
